internal/operators/manager.go

```diff
@@ -0,0 +1,159 @@
+	GenerateManifests(cluster *common.Cluster) (map[string]string, error)
+	// AnyOperatorEnabled checks whether any operator has been enabled for the given cluster
+	AnyOperatorEnabled(cluster *common.Cluster) bool
+	// GetOperatorStatus gets status of an operator of given type.
+	GetOperatorStatus(cluster *common.Cluster, operatorType models.OperatorType) string
+}
+
+// NewManager creates new instance of an Operator Manager
+func NewManager(log logrus.FieldLogger, hostApi host.API) Manager {
+	cfg := ocs.Config{}
+	err := envconfig.Process("myapp", &cfg)
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+	return NewManagerWithConfig(log, hostApi, &cfg)
+}
+
+// NewManagerWithConfig creates new instance of an Operator Manager
+func NewManagerWithConfig(log logrus.FieldLogger, hostApi host.API, cfg *ocs.Config) Manager {
+	ocsValidator := ocs.NewOCSValidator(log.WithField("pkg", "ocs-operator-state"), hostApi, cfg)
+	return Manager{
+		log:                log,
+		ocsValidatorConfig: cfg,
+		ocsValidator:       ocsValidator,
+	}
+}
+
+// GenerateManifests generates manifests for all enabled operators.
+// Returns map assigning manifest content to its desired file name
+func (mgr *Manager) GenerateManifests(cluster *common.Cluster) (map[string]string, error) {
+	lsoEnabled := false
+	ocsEnabled := mgr.checkOCSEnabled(cluster)
+	if ocsEnabled {
+		lsoEnabled = true // if OCS is enabled, LSO must be enabled by default
+	} else {
+		lsoEnabled = mgr.checkLSOEnabled(cluster)
+	}
+	operatorManifests := make(map[string]string)
+
+	if lsoEnabled {
+		manifests, err := mgr.generateLSOManifests(cluster)
+		if err != nil {
+			mgr.log.Error("Cannot generate LSO manifests due to ", err)
+			return nil, err
+		}
+		for k, v := range manifests {
+			operatorManifests[k] = v
+		}
+	}
+
+	if ocsEnabled {
+		manifests, err := mgr.generateOCSManifests(cluster)
+		if err != nil {
+			mgr.log.Error("Cannot generate OCS manifests due to ", err)
+			return nil, err
+		}
+		for k, v := range manifests {
+			operatorManifests[k] = v
+		}
+	}
+	return operatorManifests, nil
+}
+
+// AnyOperatorEnabled checks whether any operator has been enabled for the given cluster
+func (mgr *Manager) AnyOperatorEnabled(cluster *common.Cluster) bool {
+	return mgr.checkLSOEnabled(cluster) || mgr.checkOCSEnabled(cluster)
+}
+
+// ValidateOCSRequirements validates OCS requirements. Returns "true" if OCS operator is not deployed
+func (mgr *Manager) ValidateOCSRequirements(cluster *common.Cluster) bool {
+	if isEnabled(cluster, models.OperatorTypeOcs) {
+		return mgr.ocsValidator.ValidateOCSRequirements(&cluster.Cluster)
+	}
+	return true
+}
+
+// GetOperatorStatus gets status of an operator of given type.
+func (mgr *Manager) GetOperatorStatus(cluster *common.Cluster, operatorType models.OperatorType) string {
+	operator, err := findOperator(cluster, operatorType)
+	if err != nil {
+		return "Something went wrong with Unmarshalling operators"
```

Comment:
Is this first step in refactoring. Returning error would be better.

Reply:
That's also a pre-existing behavior of that code; didn't want to change it.